pkg/virt-handler/realtime_test.go

```diff
@@ -57,45 +58,45 @@ var _ = Describe("Running real time workloads", func() {
 			Entry("Fails to parse as the start range is smaller than the ending vcpu index id", "1-0", nil, fmt.Errorf("invalid mask range `1-0`")),
 			Entry("Fails to match", "1-", nil, fmt.Errorf("invalid mask value '1-' in '1-'")),
 			Entry("Fails to parse a negative vcpu ID", "-1", nil, fmt.Errorf("invalid mask value '-1' in '-1'")),
+			Entry("Fails to parse a negative vcpu ID combined with a valid index", "0,-1", nil, fmt.Errorf("invalid mask value '-1' in '0,-1'")),
 			// With range
-			Entry("Extracts a single range with one core", "0-0", map[string]maskType{"0": enabled}, nil),
-			Entry("Extracts a single range multiple cores", "0-2", map[string]maskType{"0": enabled, "1": enabled, "2": enabled}, nil),
-			Entry("Extracts a multiple non overlapping ranges", "0-2,5-6", map[string]maskType{"0": enabled, "1": enabled, "2": enabled, "5": enabled, "6": enabled}, nil),
-			Entry("Extracts a multiple overlapping ranges", "0-2,2-3", map[string]maskType{"0": enabled, "1": enabled, "2": enabled, "3": enabled}, nil),
-			Entry("Extracts a single range with double digit vcpu IDs", "10-12", map[string]maskType{"10": enabled, "11": enabled, "12": enabled}, nil),
-			Entry("Extracts a multiple range with double digit vcpu IDs", "10-12,20,21", map[string]maskType{"10": enabled, "11": enabled, "12": enabled, "20": enabled, "21": enabled}, nil),
-			Entry("Extracts a multiple range with double digit vcpu IDs spaced in between", "10-12 ,  20,21", map[string]maskType{"10": enabled, "11": enabled, "12": enabled, "20": enabled, "21": enabled}, nil),
-			Entry("Extracts a single range prefixed and postfixed with spaces", "  0-2  ", map[string]maskType{"0": enabled, "1": enabled, "2": enabled}, nil),
-			Entry("Extracts a single range with a range of 1 vcpu only", "0-0", map[string]maskType{"0": enabled}, nil),
+			Entry("Extracts a single range with one core", "0-0", cpuMask{map[string]maskType{"0": enabled}}, nil),
```

Comment:
nit: to make things a bit less verbose, you can introduce a helper function:
```go
func newMask(enabled, disabled []int) cpuMask {...}

newMask([]int{1, 2, 3}, []int{4, 5, 6})
```

Reply:
Sounds good. Will refactor :smile:

Reply:
Done.